src/components/forums/Forum.js

```diff
@@ -262,8 +357,50 @@ const Forum = () => {
                     </Card>
                 ))}
             </Box>
+            {selectedComment && (
+                <div>
+                    {/* Edit comment diaglog */}
+                    < Dialog open={isDialogOpen} onClose={handleCloseDialog} >
+                        <DialogTitle>Edit Your Comment</DialogTitle>
+                        <DialogContent>
+                            <Comment
+                                comment={comment}
+                                onEnterComment={handleEditCommentBody}
+                                commentError={commentError}
+                                fcommentErrorText={commentErrorText}
+                            />
+                        </DialogContent>
+                        <DialogActions>
+                            <Button onClick={handleCloseDialog}>Close</Button>
+                            {<Button onClick={handleEditComment}>Edit Comment</Button>}
+                        </DialogActions>
+                    </Dialog>
+                </div>
+            )}
+
+            {showEditAlertMessage && (
+                <Alert severity="success">
+                    Comment successfully edited.
+                </Alert>
+            )}
         </div>
     )
 }
+const Comment = ({ comment, onEnterComment, commentError, commentErrorText, defaultValue }) => {
+    return (
+        <Grid item>
+            <TextField
+                id="edit-comment"
+                label="Edit Comment"
+                placeholder="Enter Comment"
+                value={comment}
+                onChange={onEnterComment}
+                error={commentError}
+                fullWidth
+            />
+            <FormHelperText>{commentErrorText}</FormHelperText>
```

Comment:
seems like there is an error catch to ensure an empty comment can't be inputted, but no error appears informing the user this is the case:
![image](https://user-images.githubusercontent.com/57375371/225087125-f8ab84c5-46a4-44cd-a024-27b66093eb8b.png)
The field is red, but in our other forms, there is usually an error message below. This would be a good addition here.

Reply:
seems like you have a state above that is supposed to do this, but isn't working for some reason

Reply:
Fixed, thanks!

Reply:
nvm, when I fix this, the APi breaks and I don't know why, so for now I have removed it. Thanks!

Reply:
Update: I have fixed this issue, thank you!